### PR TITLE
Adding Full Mandatory Mexican holidays 

### PR DESCRIPTION
**Pull request by dantes4ur**
Nation-wide holidays for Mexico from the official government publication:

[https://www.gob.mx/profedet/es/articulos/sabes-cuales-son-los-dias-de-descanso-obligatorio-para-este-2024](https://www.gob.mx/profedet/es/articulos/sabes-cuales-son-los-dias-de-descanso-obligatorio-para-este-2024)

**Article 74 of the Federal Labor Law indicates the following mandatory rest days:**

I. January 1st;

II. The first Monday of February in commemoration of February 5th;

III. The third Monday of March in commemoration of March 21st;

IV. May 1st;

V. September 16th;

VI. The third Monday of November in commemoration of November 20th;

VII. October 1st every six years, when it corresponds to the transmission of the Federal Executive Power;

VIII. December 25th, and

IX. The day determined by federal and local electoral laws, in the case of ordinary elections, to conduct the electoral day.

Aside from these, any additional days are considered **optional**.

>  When dynamic holidays are incorporated into the package, we will be able to enhance and update this pull request accordingly.

### CODE REVIEW

File: src/Countries/Mexico.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+use Carbon\CarbonImmutable;
+
+class Mexico extends Country
+{
+    public function countryCode(): string
+    {
+        return 'mx';
+    }
+
+    /** @return array<string, CarbonImmutable> */
```
Comment (Nielsvanpach) [inline src/Countries/Mexico.php:14]: this can be removed

**Review by Nielsvanpach — COMMENTED** [review]
Thanks! Could you rebase this branch, in order to run the tests here?

As the Mexican public holidays seem a bit more difficult, I'll leave this PR open until some people upvoted this as approval.

File: src/Countries/Mexico.php (hunk 1 above)
Comment (dantes4ur) [reply src/Countries/Mexico.php]: The comments right? 

For that part, done

**Review by andresgaviria99 — APPROVED** [review]
LGTM

**Review by jndrdlx99 — APPROVED** [review]

**Comment by Nielsvanpach** [comment]
Thanks! Could you solve the merge conflicts?

**Comment by dantes4ur** [comment]
> Thanks! Could you solve the merge conflicts?

Already fixed, some other branch was merged with optional days.

**Comment by Nielsvanpach** [comment]
Great, thank you!

**Comment by Nielsvanpach** [comment]
I made some adjustments, could you verify if the governmentChangeDate() change is correct?
https://github.com/spatie/holidays/pull/140/files

**Comment by dantes4ur** [comment]
> I made some adjustments, could you verify if the governmentChangeDate() change is correct? https://github.com/spatie/holidays/pull/140/files

LGTM!

**Pull request merged**